pkg/asset/ignition/bootstrap/bootstrap.go

```diff
@@ -165,31 +172,35 @@ func (a *Bootstrap) Files() []*asset.File {
 }
 
 // getTemplateData returns the data to use to execute bootstrap templates.
-func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, proxy *configv1.Proxy) (*bootstrapTemplateData, error) {
+func (a *Bootstrap) getTemplateData(installConfig *types.InstallConfig, releaseImage string, releaseImageSources []types.ReleaseImageSource, proxy *configv1.Proxy) (*bootstrapTemplateData, error) {
 	etcdEndpoints := make([]string, *installConfig.ControlPlane.Replicas)
 
 	for i := range etcdEndpoints {
 		etcdEndpoints[i] = fmt.Sprintf("https://etcd-%d.%s:2379", i, installConfig.ClusterDomain())
 	}
 
-	var releaseImage string
-	if ri, ok := os.LookupEnv("OPENSHIFT_INSTALL_RELEASE_IMAGE_OVERRIDE"); ok && ri != "" {
-		logrus.Warn("Found override for ReleaseImage. Please be warned, this is not advised")
-		releaseImage = ri
-	} else {
-		var err error
-		releaseImage, err = DefaultReleaseImage()
-		if err != nil {
-			return nil, err
+	registries := []sysregistriesv2.Registry{}
+	if len(releaseImageSources) > 1 {
+		for i := 0; i < len(releaseImageSources); i++ {
+			registry := sysregistriesv2.Registry{}
+			registry.Endpoint.Location = releaseImageSources[i].Repository
+			registry.MirrorByDigestOnly = true
+			for j := 0; j < len(releaseImageSources); j++ {
+				if j == i {
```

Comment:
Non-blocking:

The M-C-O code does not have this condition; that way, the order of attempted mirrors is always the same regardless of which one was used in the pull spec. (+, yes, the `releaseImageSources[I].Repository` one is tried at the end, redundantly, but we will probably. change that in CRI-O eventually).

If you change this, please update `docs/dev/alternative_release_image_sources.md` as well.